Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSStitch1D.py

```diff
@@ -0,0 +1,421 @@
+        # 1. Scaling in y direction
+        # 2. Shift in x direction
+        # 3. Scaling in x direction
+        # 4. Shift in y direction
+        # 5. Chi^2 value
+        row0 = param.row(0).items()
+        row3 = param.row(3).items()
+
+        scale = row0[1][1]
+
+        if scale == 0.0:
+            raise RuntimeError('Fit scaling as part of stitching evaluated to zero')
+
+        # In order to determine the shift, we need to remove the scale factor
+        shift = row3[1][1] / scale
+
+        return (shift, scale)
+
+    def PyExec(self):
+        enum_map = self._make_mode_map()
+
+        mode = enum_map[self.getProperty('Mode').value]
+
+        cF = self.getProperty('HABCountsSample').value
+        cR = self.getProperty('LABCountsSample').value
+        nF = self.getProperty('HABNormSample').value
+        nR = self.getProperty('LABNormSample').value
+        q_high_angle = self._divide(cF, nF)
+        q_low_angle = self._divide(cR, nR)
+        if self.getProperty('ProcessCan').value:
+            cF_can = self.getProperty('HABCountsCan').value
+            cR_can = self.getProperty('LABCountsCan').value
+            nF_can = self.getProperty('HABNormCan').value
+            nR_can = self.getProperty('LABNormCan').value
+
+            q_high_angle_can = self._divide(cF_can, nF_can)
+            q_low_angle_can = self._divide(cR_can, nR_can)
+
+            # Now we can do the can subraction.
+            q_high_angle = self._subract(q_high_angle, q_high_angle_can)
+            q_low_angle = self._subract(q_low_angle, q_low_angle_can)
+
+        shift_factor = self.getProperty('ShiftFactor').value
+        scale_factor = self.getProperty('ScaleFactor').value
+        if not mode == Mode.NoneFit:
+
+            shift_factor, scale_factor = self._determine_factors(q_high_angle, q_low_angle, mode, scale=scale_factor,
+                                                                 shift=shift_factor)
+
+        min_q = min(min(cF.dataX(0)), min(cR.dataX(0)))
+        max_q = max(max(cF.dataX(0)), max(cR.dataX(0)))
+
+        # Crop our input workspaces
+        cF = self._crop_to_x_range(cF, min_q, max_q)
+        cR = self._crop_to_x_range(cR, min_q, max_q)
+        nF = self._crop_to_x_range(nF, min_q, max_q)
+        nR = self._crop_to_x_range(nR, min_q, max_q)
+
+        # We want: (Cf+shift*Nf+Cr)/(Nf/scale + Nr)
+        merged_q = self._calculate_merged_q(cF=cF, nF=nF, cR=cR, nR=nR, scale_factor=scale_factor,
+                                            shift_factor=shift_factor)
+
+        if self.getProperty('ProcessCan').value:
+            cF_can = self.getProperty('HABCountsCan').value
+            cR_can = self.getProperty('LABCountsCan').value
+            nF_can = self.getProperty('HABNormCan').value
+            nR_can = self.getProperty('LABNormCan').value
+
+            # Calculate merged q for the can
+            merged_q_can = self._calculate_merged_q(cF=cF_can, nF=nF_can, cR=cR_can, nR=nR_can,
```

Comment:
the merged can has a sligthly different form. See [here](https://github.com/mantidproject/mantid/blob/master/scripts/SANS/ISISCommandInterface.py#L519)